homework/index.js

```diff
@@ -29,22 +31,115 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function addRepoDetailsInfo(listElm, title, value) {
+    const trElm = createAndAppend('tr', listElm, { class: 'tr-elm' });
+    createAndAppend('td', trElm, {
+      class: 'titles',
+      text: title,
+    });
+
+    createAndAppend('td', trElm, {
+      class: 'values',
+      text: value !== null ? value : 'No Information',
+    });
+
+    return trElm;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+  function formatDate(stringDateAndTime) {
+    const dt = new Date(stringDateAndTime);
+    return dt.toLocaleString();
+  }
+
+  function sortReposByNameProp(firstRepo, secondRepo) {
+    return firstRepo.name.localeCompare(secondRepo.name);
+  }
+
+  function renderRepoDetails(repo, reposSection) {
+    const table = createAndAppend('table', reposSection, { class: 'table' });
+    const firstRow = addRepoDetailsInfo(table, 'Repository :', '');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+    });
+    addRepoDetailsInfo(table, 'Description :', repo.description);
+    addRepoDetailsInfo(table, 'Forks :', repo.forks);
+    addRepoDetailsInfo(table, 'Updated :', formatDate(repo.updated_at));
+  }
+
+  function renderContDetails(selectedRepo, contributionsUl) {
```

Comment:
I may have missed this one on my previous review, but this parameter name is incorrect. In line 97 you are passing `contributors` as the first parameter, so that parameter name should also be `contributors`. As you can see, JavaScript does not care about the names you choose, it will run your code regardless. But for humans reading your code it becomes utterly confusing is you say A but mean B.

BTW in my previous comment in this line, I made a remark about 'Cont', and I was hoping that you concluded that a better name for this function would be `renderContributorDetails`.

Reply:
I noticed after I read your feedback. I made corrections .